src/memory-store.js

```diff
@@ -15,22 +15,16 @@ class MemoryStore {
   constructor(options) {
     options = options || {};
     this._keyLimit = options.keyLimit || 100000;
-    this._keyArray = [];
-    this._store = {};
+    this._store = new Map();
 
     // Purge based on expiration every x milliseconds
     setInterval(() => {
-      let kl = this._keyArray.length;
       const now = Date.now();
-      while (kl--) {
-        const key = this._keyArray[kl];
-        if (!this._store[key]) {
-          this._keyArray.splice(kl, 1);
-          continue;
-        }
-        if (this._store[key].expiration < now) {
-          delete this._store[key];
-          this._keyArray.splice(kl, 1);
+      for (let entry of this._store.entries()) {
+        const key = entry[0];
+        const val = entry[1];
+        if (val.expiration < now) {
+          this._store.delete(key);
         }
       }
     }, options.expirationPoll || 15000);
```

Comment:
Perhaps we should disable this poll if they pass 0 for `options.expirationPoll` ?